packages/edit-site/src/components/editor/index.js

```diff
@@ -121,9 +121,18 @@ function Editor() {
 		( entitiesToSave ) => {
 			if ( entitiesToSave ) {
 				const { getEditedEntityRecord } = select( 'core' );
+				const { getTemplateInfo } = select( 'core/edit-site' );
 				entitiesToSave.forEach( ( { kind, name, key } ) => {
 					const record = getEditedEntityRecord( kind, name, key );
 
+					if ( 'postType' === kind && name === 'wp_template' ) {
+						const { title } = getTemplateInfo( record );
+						return editEntityRecord( kind, name, key, {
```

Comment:
Why do we need this? Is it basically making sure that the title never changes?

Reply:
This is because normally auto-drafts are handled in such way that they don't have titles, and therefore an auto-draft record has a blank title **regardless** of the fact that their actual entity (in the DB, returned from API, etc.) has got one.

Without that check, saving an auto-draft template would make it **lose** its title, in the actual DB I mean.
It would become like this:
<img width="927" alt="Screenshot 2020-11-18 at 11 25 51" src="https://user-images.githubusercontent.com/2070010/99524764-d4f83d80-2990-11eb-89ce-1c48ed00b14d.png">

With that check, we simply make sure that:
- If record has no title (this is the auto-draft case): keep the default title, or fallback to slug.
- Else: use the title.